backend/test_observer/controllers/test_executions/reruns.py

```diff
@@ -0,0 +1,40 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy import select
+from sqlalchemy.orm import Session, joinedload
+
+from test_observer.data_access.models import (
+    Artefact,
+    ArtefactBuild,
+    Stage,
+    TestExecution,
+    TestExecutionRerunRequest,
+)
+from test_observer.data_access.repository import get_or_create
+from test_observer.data_access.setup import get_db
+
+from .models import PendingRerun, RerunRequest
+
+router = APIRouter()
```

Comment:
Can we add `tags=["re-runs"]` to separate these in a different category in the docs view?

Reply:
Ah keep forgetting about tags

Reply:
Hmm looks like if we add rerun tag then the docs will show the endpoint twice. Once for rerun tag and once for test-executions tag:
![image](https://github.com/canonical/test_observer/assets/4087200/1f91a6c2-b8fa-4bb3-b861-db6ea920ce08)

Not sure that's desirable. We could move this out of test-executions tag, but I prefer if it's there. So how about we do without rerun tag?

Reply:
We can do without the rerun tag. I didn't realise it is under the test-executions tag, and was thinking it is actually under the default tag.